### ISSUE TITLE
Unnecessary files in the source distribution?

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/pygmt#904]
Running `python setup.py sdist` will build a PyGMT source distribution in the `dist` directory (e.g., `dist/pygmt-0.3.0.tar.gz`.

The tarball contains the PyGMT source codes. Below is the list of top-level files and directories:
```
.codecov.yml
.github
.gitignore
.pylintrc
AUTHORS.md
AUTHORSHIP.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE.txt
MAINTENANCE.md
MANIFEST.in
Makefile
PKG-INFO
README.rst
doc
environment.yml
examples
package.json
pygmt
pygmt.egg-info
pyproject.toml
requirements-dev.txt
requirements.txt
setup.cfg
setup.py
vercel.json
```

I'm not familiar with Python source distributions, but some files seem unnecessary (e.g., `vercel.json`, `package.json`, `.github`). 

Reading the [Python package guide](https://packaging.python.org/guides/using-manifest-in/), I thought most plaintext files won't be added into the source distribution, unless explicitly specified in the `MANIFEST.in` file. Perhaps my understanding is wrong.

**Comment by weiji14** [comment]
Yeah we should fix this by the next release (v0.3.1). Not that we'll reach it anytime soon, but PyPI has a 10GB total project size limit, and a 100 MB project upload limit (i.e. per release/version).

![image](https://user-images.githubusercontent.com/23487320/108116247-259e4480-7100-11eb-88de-14e61ed96176.png)

**Comment by seisman** [comment]
So setuptools_scm includes all files in the source distribution by default.

See https://github.com/pypa/setuptools_scm#file-finders-hook-makes-most-of-manifestin-unnecessary and https://github.com/pypa/setuptools_scm/issues/516.

I think we need to explicitly exclude unnecessary files in the `MANIFEST.in` file.
